packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
Okay, good to know, but I'd still like to see the `val + val => val` case work.  Since we have `append` for mutables, `add` is our primary corresponding tool for immutables.